### PR TITLE
Add tests to PythonOperator

### PR DESCRIPTION
**Pull request by vincbeck**
This work was done by [shahar1](https://github.com/shahar1) in #27797. The original PR was closed before reaching completion. Since I thought very little was left to have a working PR, I created this one

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/operators/python.py [hunk 1]
```diff
@@ -370,12 +375,8 @@ def _read_result(self, path: Path):
...
         try:
             return self.pickling_library.loads(path.read_bytes())
-        except ValueError:
-            self.log.error(
-                "Error deserializing result. Note that result deserialization "
-                "is not supported across major Python versions."
-            )
-            raise
+        except ValueError as e:
+            raise DeserializingResultError(f"Error while deserializing results. Error: {e}")
```
Comment (uranusjr) [inline airflow/operators/python.py]: Isn’t this message useless since you overriden `__str__` in the class?

Comment (vincbeck) [reply airflow/operators/python.py]: Good catch!

**Review by uranusjr — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
Tests are failing.

**Comment by vincbeck** [comment]
Trying to reproduce locally by running `breeze testing tests --backend postgres --python 3.7 tests/decorators/test_python_virtualenv.py` but unsuccessfully. Tests are passing locally using breeze but they are not in the CI

**Review by pankajastro — APPROVED** [review]

**Comment by potiuk** [comment]
> Trying to reproduce locally by running `breeze testing tests --backend postgres --python 3.7 tests/decorators/test_python_virtualenv.py` but unsuccessfully. Tests are passing locally using breeze but they are not in the CI

Try to reproduce by running the whole test type @vincbeck - this might be side effect from another test run (which means that the side-effect needs to be investigated and removed). And make sure to rebase the PR first (to see if it still fails in latest main) , and rebuild the image before running it.

**Comment by potiuk** [comment]
Also when all else fails and you cannot reproduce it - you should follow https://github.com/apache/airflow/blob/main/CI.rst#replicating-the-ci-jobs-locally -> where (after rebasing it) you will be able to download and pull the very image that was run during the CI run locally and see if you can reproduce the issue (again by runnning a complete test type rather than individually. If you can, then you can enter the image downloadeda and try to reproduce it there by running the same `pytest` command that is run in CI (You will see the exact pytest command in the failing output

**Comment by vincbeck** [comment]
Thanks @potiuk for the help! Doing it now

**Pull request merged**